### DISCUSSION TITLE
Saving Epochs with through conditions / Only lowest loss

### DISCUSSION THREAD

**Discussion by RealBeyondMaster** [bmaltais/kohya_ss#1605] [Ideas]
Hi all
It would be great  if we could  have an option to change the saving of the epochs  from "every epoch", "every 5th epoch" to
like:

**Save only if the Loss is lower than the 5th epoch, and then save only if it's lower than that previous low etc.**. This way we wouldn't overburden the HDD and only keep the best ones anyways. 
I would really appreciate it.

Further thoughts: Could it be possible  do save  epochs while it is training during an epoch, like saving during a STEP? That would be  even a finder distinction.

**Comment by swayam1998** [comment]
This is what I had been looking for. I am not sure why the option to save checkpoint at the lowest loss is not available in the UI. Is there some way to do it programmatically.
